tests/lib/rules/prefer-wait-for.js

```diff
@@ -0,0 +1,167 @@
+'use strict';
+
+const rule = require('../../../lib/rules/prefer-wait-for');
+const RuleTester = require('eslint').RuleTester;
+
+const ruleTester = new RuleTester({
+  parserOptions: { ecmaVersion: 2018, sourceType: 'module' },
+});
+ruleTester.run('prefer-wait-for', rule, {
+  valid: [
+    {
+      code: `async () => {
+        await waitFor(() => {});
+      }`,
+    },
+    {
+      code: `async () => {
+        await waitForElementToBeRemoved(() => {});
+      }`,
+    },
+    {
+      code: `async () => {
+        await waitForSomethingElse(() => {});
+      }`,
+    },
+    {
+      code: `import { wait } from 'some-testing-library'`,
```

Comment:
I just thought of this, but shouldn't these be renamed as well? 
Otherwise it wont compile, right? 🤔

Reply:
Yes, that's what I was discussing with @thomlom in a previous comment. We would need to check from where it's imported to being able to 1) check if it's the `wait` we are expecting and 2) rename the import too. Haven't done this before, so I'm checking how to do it properly.

Reply:
We could use something similar as https://github.com/cartant/eslint-plugin-rxjs/blob/master/source/rules/just.ts

Reply:
I'll give it a try later to see if I can mix that with the current rule code. Thanks.

Reply:
I'm also writing something to see how we could do it

Reply:
@Belco90 what do you think of this?
It will also take care of https://github.com/testing-library/eslint-plugin-testing-library/pull/88#discussion_r392661245, and only rename testing-library occurrences 

https://astexplorer.net/#/gist/6994663f6be1b74b6b6fb41b14320790/latest

Reply:
Ok, I'll try to put all this together with the tests to see if this works as expected. Thanks!

Reply:
👍 feel free to ping me for some help or to do another review

Reply:
Thanks! I've been little bit busy and only had time to prepare the tests, but should have some spare time today to check everything together. I'll let you know if need some help.

Reply:
There's no rush :)

Reply:
It was a tough one but I got it! Please check the final rule implementation. @timdeschryver it was inspired by your code but I had to implement the import reporter in a different way as it wasn't working properly together with everything else. Tests passing now (as pipeline not run here because doesn't match branching rules):

<img width="1442" alt="image" src="https://user-images.githubusercontent.com/2677072/77113685-8e31fc80-6a2b-11ea-8c2e-abecdef37e82.png">

Reply:
I found one case not covered: multiline imports. I'm on it :)

Reply:
Now it's fixed, all import cases covered 💪